core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcToEnumerableConverter.java

```diff
@@ -195,7 +235,14 @@ public Result implement(EnumerableRelImplementor implementor, Prefer pref) {
   }
 
   private List<ConstantExpression> toIndexesTableExpression(SqlString sqlString) {
-    return sqlString.getDynamicParameters().stream()
+    return Optional.ofNullable(sqlString.getDynamicParameters()).orElse(ImmutableList.of()).stream()
+        .map(Expressions::constant)
+        .collect(Collectors.toList());
+  }
+
+  private List<ConstantExpression> toIndexCorrelateExpression(SqlString sqlString,
+      StringBuilder querySqlSb, SqlDialect sqlDialect) {
+    return sqlDialect.getDynamicTypeIndexs(querySqlSb, sqlString.getSql()).stream()
```

Comment:
I am not sure whether `getDynamicTypeIndexs` should be in SqlDialect. 

Is this the format of SQL that this function recognizes defined in SQL standard? What kind of vendors support it? Is it a common used syntax?

Reply:
you're right, i changed the implementation